cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/validation/StaticValidatorImpl.java

```diff
@@ -0,0 +1,91 @@
+public class StaticValidatorImpl implements StaticValidator {
+
+    @Override
+    public void validateSlangFile(File slangFile, Executable executable, Metadata metadata, Boolean shouldValidateDescription){
+        validateNamespace(slangFile, executable);
+
+        validateExecutableName(slangFile, executable);
+
+        if (shouldValidateDescription) {
+            validateExecutableAgainstMetadata(executable, metadata);
+        }
+    }
+
+    private void validateExecutableAgainstMetadata(Executable executable, Metadata metadata) {
+        validateInOutParams(metadata.getInputs(), executable.getInputs(), "Error for executable " +
+                executable.getId() + ": Input");
+        validateInOutParams(metadata.getOutputs(), executable.getOutputs(), "Error for executable " +
+                executable.getId() + ": Output");
+        validateInOutParams(metadata.getResults(), executable.getResults(), "Error for executable " +
+                executable.getId() + ": Result");
+    }
+
+    private void validateInOutParams(Map<String, String> metadataInOutParams, List<? extends InOutParam> inOutParams, String errorMessagePrefix) {
+        for (InOutParam inOutParam : ListUtils.emptyIfNull(inOutParams)) {
+            if (metadataInOutParams == null) {
+                throw new RuntimeException(errorMessagePrefix + "s are missing description entirely.");
+            } else if (metadataInOutParams.get(inOutParam.getName()) == null) {
+                throw new RuntimeException(errorMessagePrefix + " '" + inOutParam.getName() + "' is missing description.");
+            }
+        }
+    }
+
+    private void validateNamespace(File slangFile, Executable executable) {
+        // Validate that the namespace is not empty
+        String namespace = executable.getNamespace();
+        Validate.notEmpty(namespace, "Error validating Slang file: \'" + slangFile.getAbsoluteFile() +
+                "\'. Namespace of slang source: \'" + executable.getName() + "\' cannot be empty.");
+
+        // Validate that the namespace matches the path of the file
+        String executableNamespacePath = namespace.replace('.', File.separatorChar);
+        String namespaceErrorMessage = "Error validating Slang file: \'" + slangFile.getAbsoluteFile() +
+                "\'. Namespace of slang source: " + executable.getName() + " is wrong.\nIt is currently \'" +
+                namespace + "\', but it should match the file path: \'" + slangFile.getPath() + "\'";
+        int indexOfLastFileSeparator = slangFile.getAbsolutePath().lastIndexOf(File.separatorChar);
+        String filePathWithoutFileName = slangFile.getAbsolutePath().substring(0, indexOfLastFileSeparator);
+        Validate.isTrue(filePathWithoutFileName.toLowerCase().endsWith(executableNamespacePath.toLowerCase()), namespaceErrorMessage);
+
+        // Validate that the namespace is composed only of abc letters, _ or -
+        Pattern pattern = Pattern.compile("^[\\w-\\.]+$");
```

Comment:
`pattern' should be static, otherwise the compilation of the regex which is time consuming is done each time